stdlib/tests/collections/smt.rs

```diff
@@ -1,664 +1,194 @@
-    let val_b = [ONE, ONE, ZERO, ZERO];
-
-    // this tests a complex insertion when a leaf moves from depth 16 to depth 48; this moves
-    // node at depth 16 to depth 48 and inserts a new node at depth 48
-    let init_smt = smt.clone();
-    smt.insert(key_b.into(), val_b);
-    let new_map_entries = [build_node_entry(key_a, val_a, 48), build_node_entry(key_b, val_b, 48)];
-    assert_set(&init_smt, key_b, EMPTY_VALUE, val_b, smt.root().into(), &new_map_entries);
-
-    // update a value under key_a; this inserts one entry into the advice map
-    let init_smt = smt.clone();
-    let val_a2 = [ONE, ZERO, ZERO, ONE];
-    smt.insert(key_a.into(), val_a2);
-    let new_map_entries = [build_node_entry(key_a, val_a2, 48)];
-    assert_set(&init_smt, key_a, val_a, val_a2, smt.root().into(), &new_map_entries);
-
-    // insert a value under key_c which has the same 32-bit prefix as A and B; this inserts
-    // one entry into the advice map
-    let raw_c = 0b00000000_00000000_11111111_11111111_00111111_01111111_11111111_11111111_u64;
-    let key_c = RpoDigest::from([ONE, ONE, ONE, Felt::new(raw_c)]);
-    let val_c = [ONE, ONE, ONE, ZERO];
-
-    let init_smt = smt.clone();
-    smt.insert(key_c.into(), val_c);
-    let new_map_entries = [build_node_entry(key_c, val_c, 48)];
-    assert_set(&init_smt, key_c, EMPTY_VALUE, val_c, smt.root().into(), &new_map_entries);
-
-    // at this point the tree has 3 nodes A, B, C, all 3 are a depth 48 and share the same 32-bit
-    // prefix; also B and C share the same 34-bit prefix.
-
-    // remove node A from the tree; since B and C share the same 34-bit prefix, they remain at
-    // depth 48
-    let init_smt = smt.clone();
-    smt.insert(key_a.into(), EMPTY_VALUE);
-    assert_set(&init_smt, key_a, val_a2, EMPTY_VALUE, smt.root().into(), &[]);
-
-    // remove node B from the tree; this will move node C to depth 16, and thus inserts a new
-    // entry into the advice map for node C at depth 16
-    let init_smt = smt.clone();
-    smt.insert(key_b.into(), EMPTY_VALUE);
-    let new_map_entries = [build_node_entry(key_c, val_c, 16)];
-    assert_set(&init_smt, key_b, val_b, EMPTY_VALUE, smt.root().into(), &new_map_entries);
+    let key = LEAVES[0].0;
+    let value = [42323_u64.into(); 4];
+    let (init_stack, final_stack, store, advice_map) = prepare_insert_or_set(key, value, &mut smt);
+    build_test!(source, &init_stack, &[], store, advice_map).expect_stack(&final_stack);
 }
 
+/// Tests inserting an empty value to an empty tree
 #[test]
-fn tsmt_set_48_from_32() {
-    let mut smt = TieredSmt::default();
-
-    // insert a value under key_a into an empty tree
-    let raw_a = 0b00000000_00000000_11111111_11111111_11111111_11111111_11111111_11111111_u64;
-    let key_a = RpoDigest::from([ONE, ONE, ONE, Felt::new(raw_a)]);
-    let val_a = [ONE, ZERO, ZERO, ZERO];
-    smt.insert(key_a.into(), val_a);
+fn test_smt_set_empty_value_to_empty_tree() {
```

Comment:
nit: should this be `test_smt_set_empty_value_to_empty_value`?

Reply:
In the test, we insert to an empty tree (see [comment](https://github.com/0xPolygonMiden/miden-vm/pull/1215#discussion_r1469858513)), which then implies that we insert to an empty leaf (they're all empty), which old empty values.

I would probably do `test_smt_set_empty_value_to_empty_leaf`? Changed it to that, but can name it whatever else too